rotkehlchen/chain/evm/decoding/interfaces.py

```diff
@@ -248,27 +233,45 @@ def reload_data(self) -> Optional[Mapping[ChecksumEvmAddress, tuple[Any, ...]]]:
             query_method=self.query_data_method,
             save_method=self.save_data_to_cache_method,
         )
-        new_pools, new_gauges = self.read_pools_and_gauges_from_cache_method()
-        if isinstance(new_pools, dict) and isinstance(self.pools, dict):  # a pool is a dict of lp token to pool addresses  # noqa: E501
-            pools_diff = set(new_pools.keys()) - set(self.pools.keys())
-        elif isinstance(new_pools, set) and isinstance(self.pools, set):
-            pools_diff = new_pools - self.pools
-        else:
-            log.error('Reloading data for pools and gauges failed because types are incorrect')
-            return None
 
-        gauges_diff = new_gauges - self.gauges
-        if len(pools_diff) == 0 and len(gauges_diff) == 0:
+        new_cache_data = self.read_data_from_cache_method()
+        cache_diff = [
+            (data.keys() if isinstance(data, dict) else data) -  # operation between dict_keys and set resolves to set  # noqa: E501
+            (new_data.keys() if isinstance(new_data, dict) else new_data)
+            for data, new_data in zip(self.cache_data, new_cache_data)
+        ]
```

Comment:
you never opearte on different types if I understood correctly

```suggestion
        cache_diff = [  # get the new items for the different information store in the cache
            (data.keys() if isinstance(data, dict) else data) -
            (new_data.keys() if isinstance(new_data, dict) else new_data)
            for data, new_data in zip(self.cache_data, new_cache_data)
        ]
```

also should be new_data - data instead of data - new_data?